src/transformStateWithClones.js

```diff
@@ -6,8 +6,33 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  let modifiedState = { ...state };
+  const result = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        modifiedState = { ...modifiedState, ...action.extraData };
+        result.push({ ...modifiedState });
```

Comment:
Pushing the modified state into the result array should be done once after the switch statement to avoid repetition and to adhere to the DRY principle.

Reply:
Agree